### PR TITLE
Fix pillar highlighting. Plus a few upgrades...

### PR DESCRIPTION
**Pull request by philmcmahon**
##  What does this change?
Pillar highlighting is currently [broken](https://www.theguardian.com/politics/2019/jul/29/johnson-refuses-to-meet-eu-leaders-unless-they-scrap-backstop?guui) on dotcom rendering as a result of the data model changes - pillars no longer have a 'pillar' property. The main interesting bit of the change is in `pillars.ts` where we're using [const assertions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#const-assertions) to restrict the type of the `allPillars` array. This then allows 'find-pillar.tsx' to return a `Pillar` rather than `Pillar | undefined` (we default to the 'news' pillar where undefined.

I've ended up upgrading typescript to support this change, and with that making some changes to babel and various testing libs.

## Why?
It looks nice

### CODE REVIEW

File: packages/frontend/amp/components/Sidebar.tsx [hunk 1]
```diff
@@ -198,15 +198,20 @@ const template = `
...
-        >
-            <template type="amp-mustache">
-                <div dangerouslySetInnerHTML={{ __html: template }} />
-            </template>
-        </amp-list>
-    </amp-sidebar>
-);
+export const Sidebar: React.FC<{ nav: NavType }> = ({ nav }) => {
+    // this next line is necessary cos react has a 'template' object with no 'type' property.
+    // By saying 'as {}' we can pretend we're not adding the 'type' property and thus avoid unhappy type errors
```
Comment (nicl) [inline packages/frontend/amp/components/Sidebar.tsx]: What is the advantage of this over the previous `template: any` override?

File: packages/frontend/amp/components/moustache.tsx [hunk 2]
```diff
@@ -20,6 +20,9 @@ export const MoustacheVariable: React.FC<{ name: string }> = ({ name }) => (
     <>{moustacheVariable(name)}</>
 );
 
-export const MoustacheTemplate: React.FC<{}> = ({ children }) => (
-    <template type="amp-mustache">{children}</template>
-);
+export const MoustacheTemplate: React.FC<{}> = ({ children }) => {
+    // this next line is necessary cos react has a 'template' object with no 'type' property.
+    // By saying 'as {}' we can pretend we're not adding the 'type' property and thus avoid unhappy type errors
+    const props = { type: 'amp-mustache' } as {};
```
Comment (nicl) [inline packages/frontend/amp/components/moustache.tsx]: Ditto.

File: packages/frontend/index.d.ts [hunk 3]
```diff
@@ -279,6 +279,9 @@ declare module 'minify-css-string' {
 
 // tslint:disable-next-line no-namespace
 declare namespace JSX {
+    interface HTMLTemplateElement extends HTMLElement {
```
Comment (nicl) [inline packages/frontend/index.d.ts]: What does this allow?

File: packages/frontend/lib/pillars.ts [hunk 4]
```diff
@@ -1,13 +1,18 @@
...
     'opinion',
     'sport',
     'culture',
     'lifestyle',
-    'labs',
-];
+] as const;
+const fakePillars = ['labs'] as const;
+
+export const allPillars = [...realPillars, ...fakePillars];
```
Comment (nicl) [inline packages/frontend/lib/pillars.ts]: Could we instead just type these using the types in index.d.ts? E.g. like:

    const allPillars: []Pillar = [..]

File: packages/frontend/amp/components/Sidebar.tsx (hunk 1 above)
Comment (philmcmahon) [reply packages/frontend/amp/components/Sidebar.tsx]: `template: any` isn't possible any more as 'template' has become a thing in react

File: packages/frontend/index.d.ts (hunk 3 above)
Comment (philmcmahon) [reply packages/frontend/index.d.ts]: oops, nothing. will remove!

File: packages/frontend/lib/pillars.ts (hunk 4 above)
Comment (philmcmahon) [reply packages/frontend/lib/pillars.ts]: I'm not sure, @AWare ?

File: packages/frontend/amp/components/Sidebar.tsx (hunk 1 above)
Comment (philmcmahon) [reply packages/frontend/amp/components/Sidebar.tsx]: I think this is as a result of the typescript or tsline upgrades, I'm not sure

**Pull request closed without merging**